### PR TITLE
Created card fields components

### PR DESCRIPTION
**Pull request by AleGastelum**
This PR introduces the following components:
- PayPalCardField
- PayPalCardNumberField
- PayPalCardExpiryField
- PayPalCardCvvField

`PayPalCardField` is the base component that actually handles field creation and rendering logic of the fields. `PayPalCardNumberField`, `PayPalCardExpiryField`, and `PayPalCardCvvField` components are just small wrapper components that wrap `PayPalCardField.tsx` component to pass the appropriate type to it, abstracting this from the merchant.

Additional to this, types in `paypal-js` where updated to account the new `destroy` methods added to Card Fields.

### CODE REVIEW

File: packages/paypal-js/types/v6/components/card-fields.d.ts [hunk 1]
```diff
@@ -153,7 +174,8 @@ type BaseCardFieldsSession = {
           .appendChild(numberField);
      * ```
      */
-    createCardFieldsComponent: (config: CardFieldOptions) => HTMLElement;
+
+    createCardFieldsComponent: (config: CardFieldOptions) => CardFieldComponent;
```
Comment (EvanReinstein) [inline packages/paypal-js/types/v6/components/card-fields.d.ts]: Will this space above the key remove the intellisense hover to show the comment?

Comment (AleGastelum) [reply packages/paypal-js/types/v6/components/card-fields.d.ts]: It seems to keep working with the space, but I will remove it.

File: packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * // Basic usage creating a CVV field
+ * <PayPalCardCvvField
+ *   placeholder="Enter CVV"
+ *   containerStyles={{ height: "3rem", marginBottom: "1rem" }}
+ * />
+ */
+export const PayPalCardCvvField = (
+    options: PayPalCardCvvFieldProps,
+): JSX.Element | null => {
+    return <PayPalCardField type="cvv" {...options} />;
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx]: I've learned recently that the spread operator negates some typescript functionality, for example using a spread operator can negate a discriminating union. Can you double check that the options are typed correctly?

Also, I am concerned that `options` is a non-primitive type prop and will trigger a re-render each time even though none of the options have changed. In other places we use `deepCompareMemoize` to prevent this. Let me know what you think

File: packages/react-paypal-js/src/v6/components/PayPalCardExpiryField.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * @example
+ * // Basic usage creating an expiry field
+ * <PayPalCardExpiryField
+ *   placeholder="Enter an expiry date"
+ *   containerStyles={{ height: "3rem", marginBottom: "1rem" }}
+ * />
+ */
+export const PayPalCardExpiryField = (
+    options: PayPalCardExpiryFieldProps,
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/components/PayPalCardExpiryField.tsx]: Similar `options` non-primitive comment as above, as well as spread operator curiousity. 👍

File: packages/react-paypal-js/src/v6/components/PayPalCardField.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+            cardFieldContainerRef.current?.appendChild(cardField);
+        } catch (error) {
+            setError(toError(error));
+        }
+
+        return () => {
+            cardField?.destroy();
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [cardFieldsSession]);
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/components/PayPalCardField.tsx:56]: Is this useEffect ignoring `options` updates? Should merchants be able to update the field?

File: .changeset/cold-regions-rescue.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+"@paypal/react-paypal-js": patch
```
Comment (EvanReinstein) [inline .changeset/cold-regions-rescue.md]: Would you mind updating this to `minor`?

File: .changeset/quick-chefs-occur.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"@paypal/paypal-js": patch
```
Comment (EvanReinstein) [inline .changeset/quick-chefs-occur.md]: Same comment as above, please update to `minor`

File: packages/paypal-js/types/v6/components/card-fields.d.ts [hunk 7]
```diff
@@ -135,6 +135,27 @@ export type CardFieldOptions = {
...
 
+export type CardFieldComponent = HTMLElement & {
+    /**
+     * Use this method to destroy the Card Field component and clean up any associated resources. After calling this method, the Card Field component will no longer be usable.
+     * 
+     * @example
+     * ```typescript
+     * const numberField = cardFieldsInstance.createCardFieldsComponent({
+             type: "number",
+            placeholder: "Enter a number",
```
Comment (EvanReinstein) [inline packages/paypal-js/types/v6/components/card-fields.d.ts]:
```suggestion
            type: "number",
            placeholder: "Enter a number",
```

File: packages/react-paypal-js/src/v6/components/PayPalCardNumberField.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ * // Basic usage creating a number field
+ * <PayPalCardNumberField
+ *   placeholder="Enter a card number"
+ *   containerStyles={{ height: "3rem", marginBottom: "1rem" }}
+ * />
+ */
+export const PayPalCardNumberField = (
+    options: PayPalCardNumberFieldProps,
+): JSX.Element | null => {
+    return <PayPalCardField type="number" {...options} />;
```
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/components/PayPalCardNumberField.tsx]: Same comment re: non-primitive `options` and the spread operator 👍

File: packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx (hunk 2 above)
Comment (EvanReinstein) [reply packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx]: Ok, finally got to `PayPalCardField` and saw the Ref there!

File: packages/react-paypal-js/src/v6/components/PayPalCardField.test.tsx
Comment (EvanReinstein) [inline packages/react-paypal-js/src/v6/components/PayPalCardField.test.tsx:1]: I am pretty sure there is a missing test case here, for when the session transitions from non-null to null while the component stays mounted (when sdkInstance changes and the provider resets). 

We should verify that destroy() is called on the existing field and the container is empty afterward.

File: .changeset/cold-regions-rescue.md (hunk 5 above)
Comment (AleGastelum) [reply .changeset/cold-regions-rescue.md]: Updated!

File: .changeset/quick-chefs-occur.md (hunk 6 above)
Comment (AleGastelum) [reply .changeset/quick-chefs-occur.md]: Updated as well!

File: packages/react-paypal-js/src/v6/components/PayPalCardNumberField.tsx (hunk 8 above)
Comment (AleGastelum) [reply packages/react-paypal-js/src/v6/components/PayPalCardNumberField.tsx]: Removed the use of the spread operator and made it more explicit

File: packages/react-paypal-js/src/v6/components/PayPalCardExpiryField.tsx (hunk 3 above)
Comment (AleGastelum) [reply packages/react-paypal-js/src/v6/components/PayPalCardExpiryField.tsx]: Removed the use of the spread operator and made it more explicit

File: packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx (hunk 2 above)
Comment (AleGastelum) [reply packages/react-paypal-js/src/v6/components/PayPalCardCvvField.tsx]: I removed the use of the spread operator and made it more explicit.

For the options concern, this current Card Fields implementation won't trigger re-renders if the options change, as the `PayPalCardField` component stores that object in a ref. Current implementation won't react if options object change after field creation.

File: packages/react-paypal-js/src/v6/components/PayPalCardField.tsx (hunk 4 above)
Comment (AleGastelum) [reply packages/react-paypal-js/src/v6/components/PayPalCardField.tsx:56]: That is correct, current implementation is ignoring `option` updates, as updating options after field creation is not a feature the Card Fields Web SDK provides currently.

File: packages/react-paypal-js/src/v6/components/PayPalCardField.test.tsx
Comment (AleGastelum) [reply packages/react-paypal-js/src/v6/components/PayPalCardField.test.tsx:1]: Good call, I added the missing test named `"should call field's destroy method if session becomes null"`
